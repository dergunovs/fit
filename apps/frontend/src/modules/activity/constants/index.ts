import { TActivityChartType } from 'fitness-tracker-contracts';

export const URL_ACTIVITY = '/activities';
export const URL_ACTIVITY_CREATE = `${URL_ACTIVITY}/create`;
export const URL_ACTIVITY_EDIT = `${URL_ACTIVITY}/edit`;

export const URL_ACTIVITY_ADMIN = '/activities_admin';
export const URL_ACTIVITY_ADMIN_EDIT = `${URL_ACTIVITY_ADMIN}/edit`;

export const ACTIVITY_LIST_HEADERS = [{ title: 'Создано' }, { title: 'Пользователь' }];

export const CHART_TYPES: { title: string; value: TActivityChartType }[] = [
  { title: 'Занятия', value: 'activity' },
  { title: 'Подходы', value: 'set' },
  { title: 'Повторы', value: 'repeat' },
  { title: 'Мышцы', value: 'muscle' },
];

export const ACTIVITY_STATISTICS_GAP = 30;
