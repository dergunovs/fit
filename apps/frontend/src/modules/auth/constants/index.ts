export const URL_SETUP = '/setup';

export const TOKEN_NAME = 'fit_token';

export const AUTH_FORM_HEADER_LOGIN = 'Логин';
export const AUTH_FORM_HEADER_SETUP = 'Создать администратора';

export const AUTH_FORM_SUBMIT_BUTTON_LOGIN = 'Войти';
export const AUTH_FORM_SUBMIT_BUTTON_SETUP = 'Создать';
