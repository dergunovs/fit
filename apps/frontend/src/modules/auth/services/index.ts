import {
  API_AUTH_GET,
  API_AUTH_LOGIN,
  API_AUTH_REGISTER,
  API_AUTH_SETUP,
  API_AUTH_CONFIRM,
  API_AUTH_RESET,
  TGetAuthDTO,
  TPostAuthConfirmTokenDTO,
  TPostAuthConfirmTokenDataDTO,
  TPostAuthLoginDTO,
  TPostAuthLoginDataDTO,
  TPostAuthRegisterDTO,
  TPostAuthRegisterDataDTO,
  TPostAuthSetupDTO,
  TPostAuthSetupDataDTO,
  TPostAuthResetPasswordDataDTO,
  TPostAuthResetPasswordDTO,
  TPostAuthRegisterQueryDTO,
  TPostAuthResetPasswordQueryDTO,
} from 'fitness-tracker-contracts';
import { useQuery, useMutation, api } from 'mhz-helpers';

export const authService = {
  check: (options: object) => {
    return useQuery({
      queryKey: [API_AUTH_GET],
      queryFn: async () => {
        const { data } = await api.get<TGetAuthDTO>(API_AUTH_GET);

        return data;
      },
      ...options,
    });
  },

  login: (options: object) => {
    return useMutation({
      mutationKey: [API_AUTH_LOGIN],
      mutationFn: async (formData: TPostAuthLoginDataDTO) => {
        const { data } = await api.post<TPostAuthLoginDTO>(API_AUTH_LOGIN, formData);

        return data;
      },
      ...options,
    });
  },

  setup: (options: object) => {
    return useMutation({
      mutationKey: [API_AUTH_SETUP],
      mutationFn: async (formData: TPostAuthSetupDataDTO) => {
        const { data } = await api.post<TPostAuthSetupDTO>(API_AUTH_SETUP, formData);

        return data;
      },
      ...options,
    });
  },

  register: (lang: string, options: object) => {
    return useMutation({
      mutationKey: [API_AUTH_REGISTER],
      mutationFn: async (formData: TPostAuthRegisterDataDTO) => {
        const params: TPostAuthRegisterQueryDTO = { lang };

        const { data } = await api.post<TPostAuthRegisterDTO>(API_AUTH_REGISTER, formData, { params });

        return data;
      },
      ...options,
    });
  },

  confirmToken: (options: object) => {
    return useMutation({
      mutationKey: [API_AUTH_CONFIRM],
      mutationFn: async (token: TPostAuthConfirmTokenDataDTO) => {
        const { data } = await api.post<TPostAuthConfirmTokenDTO>(API_AUTH_CONFIRM, token);

        return data;
      },
      ...options,
    });
  },

  resetPassword: (lang: string, options: object) => {
    return useMutation({
      mutationKey: [API_AUTH_RESET],
      mutationFn: async (email: TPostAuthResetPasswordDataDTO) => {
        const params: TPostAuthResetPasswordQueryDTO = { lang };

        const { data } = await api.post<TPostAuthResetPasswordDTO>(API_AUTH_RESET, email, { params });

        return data;
      },
      ...options,
    });
  },
};
