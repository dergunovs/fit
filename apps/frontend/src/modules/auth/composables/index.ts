import { ref, watch, readonly } from 'vue';
import { getCookieToken, setAuth, setAuthHeader } from 'mhz-helpers';

import { authService } from '@/auth/services';
import { TOKEN_NAME } from '@/auth/constants';

const isAdminStore = ref(false);
const isAuthCheckedStore = ref(false);

export function setAdmin(isAdmin: boolean) {
  isAdminStore.value = isAdmin;
}

export const isAdmin = readonly(isAdminStore);
export const isAuthChecked = readonly(isAuthCheckedStore);

export function useAuthCheck() {
  const cookieToken = getCookieToken(TOKEN_NAME);

  if (cookieToken) {
    setAuthHeader(cookieToken);
  } else {
    isAuthCheckedStore.value = true;
  }

  const { data: user, isSuccess } = authService.check({ enabled: !!cookieToken });

  watch(
    () => user.value,
    () => {
      if (user.value?._id) {
        setAuth(true);
        isAuthCheckedStore.value = isSuccess.value;
        if (user.value.role === 'admin') setAdmin(true);
      }
    }
  );

  return { user, isAdmin, isAuthChecked };
}
