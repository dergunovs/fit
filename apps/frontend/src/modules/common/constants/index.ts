import IconExercise from '@/common/icons/nav-exercise.svg?component';
import IconActivity from '@/common/icons/nav-activity.svg?component';
import IconEquipment from '@/common/icons/weight.svg?component';
import IconUser from '@/common/icons/nav-user.svg?component';

import { URL_USER } from '@/user/constants';
import { URL_EXERCISE } from '@/exercise/constants';
import { URL_ACTIVITY_ADMIN } from '@/activity/constants';
import { URL_EQUIPMENT } from '@/equipment/constants';
import { INavItem } from '@/common/interface';
import { URL_SETUP } from '@/auth/constants';

export const NAV_ITEMS: INavItem[] = [
  { _id: '1', url: URL_EXERCISE, title: 'Упражнения', icon: IconExercise },
  { _id: '2', url: URL_ACTIVITY_ADMIN, title: 'Занятия', icon: IconActivity },
  { _id: '3', url: URL_EQUIPMENT, title: 'Оборудование', icon: IconEquipment },
  { _id: '4', url: URL_USER, title: 'Пользователи', icon: IconUser },
];

export const UPDATE_BUTTON_TEXT = 'Обновить';
export const CREATE_BUTTON_TEXT = 'Создать';

export const URL_HOME = '/';
export const URL_ERROR = '/404';

export const GUEST_PAGES = [URL_HOME, URL_SETUP];
