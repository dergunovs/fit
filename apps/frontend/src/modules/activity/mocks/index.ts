import { ref, computed } from 'vue';
import { vi } from 'vitest';
import { TDeleteActivityDTO, TGetActivitiesDTO } from 'fitness-tracker-contracts';

import * as activityComposables from '@/activity/composables';
import { activityService } from '@/activity/services';
import {
  ACTIVITIES_CALENDAR_FIXTURE,
  ACTIVITIES_FIXTURE,
  ACTIVITIES_STATISTICS_FIXTURE,
  ACTIVITY_FIXTURE,
  ACTIVITIES_CHART_FIXTURE,
} from '@/activity/fixtures';
import { mockMutationReply, mockQueryReply } from '@/common/mocks';
import { IOnSuccess } from '@/common/interface';

const dateFrom = ref('2024-11-24T21:00:00.000Z');
const dateTo = ref('2025-01-05T20:59:59.000Z');
const isDatesReady = ref(true);
const spyUpdateDates = vi.fn();

const spyGetActivity = vi.spyOn(activityService, 'getOne').mockReturnValue(mockQueryReply(ACTIVITY_FIXTURE));

const getActivitiesData: TGetActivitiesDTO = { data: ACTIVITIES_FIXTURE, total: ACTIVITIES_FIXTURE.length };

const spyGetActivities = vi
  .spyOn(activityService, 'getMany')
  .mockImplementation(() => mockQueryReply(getActivitiesData));

const spyDeleteActivity = vi.fn();

const onSuccess: IOnSuccess = {
  delete: undefined,
};

vi.spyOn(activityService, 'delete').mockImplementation((options: { onSuccess?: () => Promise<void> }) => {
  if (options.onSuccess) onSuccess.delete = options.onSuccess;

  return mockMutationReply<TDeleteActivityDTO, string>(spyDeleteActivity);
});

const spyUseActivityCalendar = vi.spyOn(activityComposables, 'useActivityCalendar').mockReturnValue({
  dateFrom,
  dateTo,
  isDatesReady,
  updateDates: spyUpdateDates,
});

const computedActivityCalendarEvents = computed(() =>
  activityComposables.convertActivityCalendarEvents(ACTIVITIES_CALENDAR_FIXTURE)
);

const spyComputedActivityCalendarEvents = vi
  .spyOn(activityComposables, 'computedActivityCalendarEvents')
  .mockImplementation(() => computedActivityCalendarEvents);

const spyGetActivitiesCalendar = vi
  .spyOn(activityService, 'getCalendar')
  .mockImplementation(() => mockQueryReply(ACTIVITIES_CALENDAR_FIXTURE));

const spyGetActivitiesStatistics = vi
  .spyOn(activityService, 'getStatistics')
  .mockImplementation(() => mockQueryReply(ACTIVITIES_STATISTICS_FIXTURE));

const spyGetActivitiesChart = vi
  .spyOn(activityService, 'getChart')
  .mockImplementation(() => mockQueryReply(ACTIVITIES_CHART_FIXTURE));

export {
  spyGetActivities,
  spyGetActivity,
  spyGetActivitiesStatistics,
  spyGetActivitiesChart,
  spyDeleteActivity,
  spyUseActivityCalendar,
  spyComputedActivityCalendarEvents,
  spyGetActivitiesCalendar,
  spyUpdateDates,
  dateFrom,
  dateTo,
  isDatesReady,
  computedActivityCalendarEvents,
  getActivitiesData,
  onSuccess,
};
