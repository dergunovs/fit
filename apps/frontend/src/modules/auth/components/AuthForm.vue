<template>
  <div>
    <h2 data-test="auth-form-header">{{ header }}</h2>

    <form @submit.prevent="submit" :class="$style.form" data-test="auth-form">
      <UiField label="Электронная почта" isRequired :error="error('email')">
        <UiInput v-model="formData.email" data-test="auth-form-email" />
      </UiField>

      <UiField label="Пароль" isRequired :error="error('password')">
        <UiInput v-model="formData.password" type="password" data-test="auth-form-password" />
      </UiField>

      <UiButton type="submit" data-test="auth-form-submit-button">{{ submitButton }}</UiButton>
    </form>
  </div>
</template>

<script setup lang="ts">
import { ref, computed } from 'vue';
import { useRouter } from 'vue-router';
import { UiButton, UiField, UiInput, toast } from 'mhz-ui';
import { useAuth, setAuthHeader, useValidator, required, email } from 'mhz-helpers';
import { IAuthData, TPostAuthLoginDTO } from 'fitness-tracker-contracts';

import { authService } from '@/auth/services';
import { URL_HOME } from '@/common/constants';
import { URL_ACTIVITY_CREATE } from '@/activity/constants';
import {
  TOKEN_NAME,
  AUTH_FORM_HEADER_SETUP,
  AUTH_FORM_HEADER_LOGIN,
  AUTH_FORM_SUBMIT_BUTTON_SETUP,
  AUTH_FORM_SUBMIT_BUTTON_LOGIN,
} from '@/auth/constants';

interface IProps {
  isSetup?: boolean;
}

const props = defineProps<IProps>();
const emit = defineEmits<{ login: [] }>();

const router = useRouter();

const { auth } = useAuth();

const formData = ref<IAuthData>({
  email: '',
  password: '',
});

const header = computed(() => (props.isSetup ? AUTH_FORM_HEADER_SETUP : AUTH_FORM_HEADER_LOGIN));
const submitButton = computed(() => (props.isSetup ? AUTH_FORM_SUBMIT_BUTTON_SETUP : AUTH_FORM_SUBMIT_BUTTON_LOGIN));

const { mutate: mutateLogin } = authService.login({
  onSuccess: (user: TPostAuthLoginDTO) => {
    if (!user.token) return;

    toast.success(`Добро пожаловать, ${user.name}!`);
    auth(user.token, setAuthHeader, TOKEN_NAME);
    emit('login');
    router.push(URL_ACTIVITY_CREATE);
  },
});

const { mutate: mutateSetup } = authService.setup({
  onSuccess: () => {
    toast.success('Администратор создан!');
    router.push(URL_HOME);
  },
});

const rules = computed(() => {
  return {
    email: [required('ru'), email('ru')],
    password: required('ru'),
  };
});

const { error, isValid } = useValidator(formData, rules);

function submit() {
  if (!isValid()) return;

  if (props.isSetup) {
    mutateSetup(formData.value);
  } else {
    mutateLogin(formData.value);
  }
}
</script>

<style module lang="scss">
.form {
  display: flex;
  flex-direction: column;
  gap: 24px;
  min-width: 300px;
}
</style>
