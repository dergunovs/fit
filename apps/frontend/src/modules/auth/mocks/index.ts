import { ref } from 'vue';
import { vi } from 'vitest';
import {
  TLocale,
  TPostAuthConfirmTokenDataDTO,
  TPostAuthConfirmTokenDTO,
  TPostAuthLoginDataDTO,
  TPostAuthLoginDTO,
  TPostAuthRegisterDataDTO,
  TPostAuthRegisterDTO,
  TPostAuthResetPasswordDataDTO,
  TPostAuthResetPasswordDTO,
  TPostAuthSetupDataDTO,
  TPostAuthSetupDTO,
} from 'fitness-tracker-contracts';

import { authService } from '@/auth/services';
import { mockMutationReply, mockQueryReply } from '@/common/mocks';
import * as authComposables from '@/auth/composables';
import { IOnSuccess } from '@/common/interface';
import { USER_FIXTURE } from '@/user/fixtures';

const mockIsAdmin = ref(true);
const mockIsAuthChecked = ref(true);

const spyUseAuthCheck = vi
  .spyOn(authComposables, 'useAuthCheck')
  .mockReturnValue({ user: ref(USER_FIXTURE), isAdmin: ref(mockIsAdmin), isAuthChecked: ref(mockIsAuthChecked) });

const spyLogin = vi.fn();
const spySetup = vi.fn();
const spyRegister = vi.fn();
const spyConfirmToken = vi.fn();
const spyResetPassword = vi.fn();

const mockOnSuccess: IOnSuccess = {
  check: undefined,
  login: undefined,
  setup: undefined,
  register: undefined,
  confirmToken: undefined,
  resetPassword: undefined,
};

const spyCheckAuth = vi.spyOn(authService, 'check').mockImplementation(() => mockQueryReply(USER_FIXTURE));

vi.spyOn(authService, 'login').mockImplementation(
  (options: { onSuccess?: (user: TPostAuthLoginDTO) => Promise<void> }) => {
    if (options.onSuccess) mockOnSuccess.login = options.onSuccess;

    return mockMutationReply<TPostAuthLoginDTO, TPostAuthLoginDataDTO>(spyLogin);
  }
);

vi.spyOn(authService, 'setup').mockImplementation((options: { onSuccess?: () => Promise<void> }) => {
  if (options.onSuccess) mockOnSuccess.setup = options.onSuccess;

  return mockMutationReply<TPostAuthSetupDTO, TPostAuthSetupDataDTO>(spySetup);
});

vi.spyOn(authService, 'register').mockImplementation((_lang: TLocale, options: { onSuccess?: () => Promise<void> }) => {
  if (options.onSuccess) mockOnSuccess.register = options.onSuccess;

  return mockMutationReply<TPostAuthRegisterDTO, TPostAuthRegisterDataDTO>(spyRegister);
});

vi.spyOn(authService, 'confirmToken').mockImplementation((options: { onSuccess?: () => Promise<void> }) => {
  if (options.onSuccess) mockOnSuccess.confirmToken = options.onSuccess;

  return mockMutationReply<TPostAuthConfirmTokenDTO, TPostAuthConfirmTokenDataDTO>(spyConfirmToken);
});

vi.spyOn(authService, 'resetPassword').mockImplementation(
  (_lang: TLocale, options: { onSuccess?: () => Promise<void> }) => {
    if (options.onSuccess) mockOnSuccess.resetPassword = options.onSuccess;

    return mockMutationReply<TPostAuthResetPasswordDTO, TPostAuthResetPasswordDataDTO>(spyResetPassword);
  }
);

export {
  spyUseAuthCheck,
  spyCheckAuth,
  spyLogin,
  spySetup,
  spyRegister,
  spyConfirmToken,
  spyResetPassword,
  mockOnSuccess,
  mockIsAdmin,
};
