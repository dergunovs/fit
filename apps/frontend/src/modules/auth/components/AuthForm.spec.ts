import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { dataTest } from 'mhz-helpers';
import { API_ACTIVITY_CALENDAR, API_ACTIVITY_CHART, API_ACTIVITY_STATISTICS } from 'fitness-tracker-contracts';

import AuthForm from './AuthForm.vue';

import { wrapperFactory } from '@/common/test';
import { mockOnSuccess, spyLogin, spySetup } from '@/auth/mocks';
import {
  spyAuth,
  spySetAuthHeaders,
  spyRouterPush,
  spyToastSuccess,
  mockIsValid,
  spyRefetchQueries,
} from '@/common/mocks';
import { URL_HOME } from '@/common/constants';
import { TOKEN_FIXTURE } from '@/auth/fixtures';
import {
  TOKEN_NAME,
  AUTH_FORM_HEADER_SETUP,
  AUTH_FORM_HEADER_LOGIN,
  AUTH_FORM_SUBMIT_BUTTON_SETUP,
  AUTH_FORM_SUBMIT_BUTTON_LOGIN,
} from '@/auth/constants';

const EMAIL = 'a@b.ru';
const PASSWORD = 'qwerty';

const ID = 'vf9gwr9bunrwber';
const NAME = 'Иван';

const form = dataTest('auth-form');
const formHeader = dataTest('auth-form-header');
const formEmail = dataTest('auth-form-email');
const formPassword = dataTest('auth-form-password');
const formSubmitButton = dataTest('auth-form-submit-button');

let wrapper: VueWrapper<InstanceType<typeof AuthForm>>;

beforeEach(() => {
  wrapper = wrapperFactory(AuthForm, { isSetup: false });
});

enableAutoUnmount(afterEach);

describe('AuthForm', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(AuthForm)).toBeTruthy();
  });

  it('shows form header and submit button text', async () => {
    expect(wrapper.find(formHeader).text()).toBe(AUTH_FORM_HEADER_LOGIN);
    expect(wrapper.find(formSubmitButton).text()).toBe(AUTH_FORM_SUBMIT_BUTTON_LOGIN);

    await wrapper.setProps({ isSetup: true });

    expect(wrapper.find(formHeader).text()).toBe(AUTH_FORM_HEADER_SETUP);
    expect(wrapper.find(formSubmitButton).text()).toBe(AUTH_FORM_SUBMIT_BUTTON_SETUP);
  });

  it('uses validation', async () => {
    mockIsValid.value = false;

    await wrapper.find(form).trigger('submit');

    expect(spyLogin).toBeCalledTimes(0);

    mockIsValid.value = true;
  });

  it('handles login by form submit', async () => {
    expect(spyLogin).toBeCalledTimes(0);
    expect(spyToastSuccess).toBeCalledTimes(0);
    expect(spyAuth).toBeCalledTimes(0);
    expect(wrapper.emitted()).not.toHaveProperty('login');
    expect(spyRouterPush).toBeCalledTimes(0);

    await wrapper.findComponent(formEmail).setValue(EMAIL);
    await wrapper.findComponent(formPassword).setValue(PASSWORD);

    await wrapper.find(form).trigger('submit');

    expect(spyLogin).toBeCalledTimes(1);
    expect(spyLogin).toBeCalledWith({ email: EMAIL, password: PASSWORD });

    await mockOnSuccess.login?.({ user: { _id: ID, email: EMAIL, name: NAME }, token: TOKEN_FIXTURE });

    expect(spyToastSuccess).toBeCalledTimes(1);

    expect(spyAuth).toBeCalledTimes(1);
    expect(spyAuth).toBeCalledWith(TOKEN_FIXTURE, spySetAuthHeaders, TOKEN_NAME);

    expect(wrapper.emitted('login')).toHaveLength(1);

    expect(spyRefetchQueries).toBeCalledTimes(3);
    expect(spyRefetchQueries).toBeCalledWith({ queryKey: [API_ACTIVITY_STATISTICS] });
    expect(spyRefetchQueries).toBeCalledWith({ queryKey: [API_ACTIVITY_CALENDAR] });
    expect(spyRefetchQueries).toBeCalledWith({ queryKey: [API_ACTIVITY_CHART] });
  });

  it('handles setup by form submit', async () => {
    await wrapper.setProps({ isSetup: true });

    expect(spySetup).toBeCalledTimes(0);
    expect(spyToastSuccess).toBeCalledTimes(0);
    expect(spyRouterPush).toBeCalledTimes(0);

    await wrapper.findComponent(formEmail).setValue(EMAIL);
    await wrapper.findComponent(formPassword).setValue(PASSWORD);

    await wrapper.find(form).trigger('submit');

    expect(spySetup).toBeCalledTimes(1);
    expect(spySetup).toBeCalledWith({ email: EMAIL, password: PASSWORD });

    await mockOnSuccess.setup?.();

    expect(spyToastSuccess).toBeCalledTimes(1);

    expect(spyRouterPush).toBeCalledTimes(1);
    expect(spyRouterPush).toBeCalledWith(URL_HOME);
  });
});
