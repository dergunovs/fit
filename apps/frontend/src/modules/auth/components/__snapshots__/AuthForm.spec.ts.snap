// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AuthForm > matches snapshot 1`] = `
"<div column="">
  <h2 data-test="auth-form-header">Логин</h2>
  <form class="_form_39c91b" data-test="auth-form">
    <fieldset label="Электронная почта" isrequired=""><input type="email" modelvalue="" autocomplete="username" data-test="auth-form-email"></fieldset>
    <div>
      <fieldset label="Пароль" isrequired=""><input type="text" modelvalue="" ispassword="" autocomplete="current-password" data-test="auth-form-password"></fieldset><button type="button" layout="plain" data-test="auth-form-password-reset-button"> Забыли пароль? </button>
      <!--v-if-->
    </div><button type="submit" data-test="auth-form-submit-button">Войти</button>
  </form>
</div>"
`;
